setup.py

```diff
@@ -24,8 +24,7 @@
     # pip>=21.3 required for editable pyproject.toml + setup.cfg based install support
     "pip>=21.3",
     "ply~=3.0",
-    # lower bound because of pydantic/pydantic#5821
-    "pydantic>=1.10.8,<2",
+    "pydantic~=2.4",
```

Comment:
Tests have been run against both versions so I see no reason to require 2.5.1 specifically.

Reply:
The purpose of `requirements.txt` vs `setup.py` wasn't clear in my mind. From what I gathered:
- `setup.py` is used for distribution when packaging the core package so we want the lowest compatible bound to have the widest compatibility possible
- `requirements.txt` is used during development when working on the core package so we want the latest available version. (Dependabot makes sure of this)

----
Is this correct ? And related question, is it possible that having different version in both files lead to inconsistencies ?

Reply:
This is indeed correct. `requirements.txt` is what we call a lock file. It locks dependencies to a specific version, but only when explicitly requested with `-r`. It is used for repetitive testing. `setup.py`, `setup.cfg` or `pyproject.toml` specify the project's dependencies. These are also referred to (at least by dependabot) as the manifest files. They represent the constraints that have to be satisfied to install the package, be it from source or from PyPi. No manual `-r` is required for them to take effect.

> is it possible that having different version in both files lead to inconsistencies

Depends how you install. If you do it with one command (`pip install -e . -r requirements.txt`), this would fail, so it would never be merged. I think tox does it like that.